tests/validations/mta/mtaValidations.spec.ts

```diff
@@ -0,0 +1,284 @@
+        testVscode.languages,
+        "createDiagnosticCollection"
+      );
+
+      await validateWsMtaYamls(disposables);
+
+      expect(createDiagnosticCollectionSpy.callCount).to.equal(0);
+    });
+
+    it("returns no diagnostics when mta.yaml and dev.mtaext have no errors in the WS", async () => {
+      sinon.stub(fsExtra, "pathExists").resolves(true);
+      sinon.stub(Mta.prototype, "validate").resolves({});
+
+      await testValidateWsMtaYamls([]);
+    });
+
+    it("returns diagnostics when mta.yaml and dev.mtaext have errors in the WS", async () => {
+      const expectedEntries: [
+        Uri,
+        Diagnostic[] | undefined
+      ][] = prepDiagnostics();
+      await testValidateWsMtaYamls(expectedEntries);
+    });
+
+    async function testValidateWsMtaYamls(
+      expectedCollectionEntries: [Uri, Diagnostic[] | undefined][]
+    ) {
+      const fileName = "mtaValidations.spec.js";
+      const uriPath = resolve(__dirname, fileName);
+      const uri = {
+        fsPath: uriPath,
+      } as Uri;
+      sinon
+        .stub(testVscode.workspace, "findFiles")
+        .returns(Promise.resolve([uri]));
+
+      let collectionEntries:
+        | [Uri, Diagnostic[] | undefined][]
+        | undefined = undefined;
+      sinon.stub(testVscode.languages, "createDiagnosticCollection").returns({
+        forEach: () => {
+          return;
+        },
+        set: (newCollectionEntries: [Uri, Diagnostic[] | undefined][]) => {
+          collectionEntries = newCollectionEntries;
+        },
+      });
+
+      await validateWsMtaYamls(disposables);
+      expect(collectionEntries).to.deep.equal(expectedCollectionEntries);
+    }
+  });
+
+  function prepDiagnostics() {
+    const mtaPath = resolve(__dirname, MTA_YAML);
+    const devMtaExtPath = resolve(__dirname, DEV_MTA_EXT);
+
+    const validationResult: Record<string, mta.Issue[]> = {
+      [mtaPath]: [
+        {
+          severity: "warning",
+          message: 'mapping key "_schema-version" already defined at line 1',
+          line: 2,
+          column: 0,
+        },
+      ],
+      [devMtaExtPath]: [
+        {
+          severity: "warning",
+          message: 'mapping key "_schema-version" already defined at line 3',
+          line: 4,
+          column: 0,
+        },
+      ],
+    };
+
+    const firstCallRange = {
+      start: {
+        line: 1,
+        character: 0,
+      },
+      end: {
+        line: 1,
+        character: 0,
+      },
+    } as Range;
+    const secondCallRange = {
+      start: {
+        line: 3,
+        character: 0,
+      },
+      end: {
+        line: 3,
+        character: 0,
+      },
+    } as Range;
+    const firstCallUri = {
+      fsPath: mtaPath,
+    } as Uri;
+    const secondCallUri = {
+      fsPath: devMtaExtPath,
+    } as Uri;
+
+    sinon.stub(fsExtra, "pathExists").resolves(true);
+    sinon.stub(Mta.prototype, "validate").resolves(validationResult);
+    sinon
```

Comment:
Why do you need to stub this? Won't it work without the stub?

Reply:
done